vllm/model_executor/layers/quantization/fp8.py

```diff
@@ -231,9 +231,13 @@ def apply(self,
         # ops.scaled_fp8_quant supports both dynamic and static quant.
         #   If dynamic, layer.act_scale is None and x_scale computed from x.
         #   If static,  layer.act_scale is scalar and x_scale set to act_scale.
-        qinput, x_scale = ops.scaled_fp8_quant(x, layer.act_scale)
+        qinput, x_scale = ops.scaled_fp8_quant(x,
+                                               layer.act_scale,
+                                               batch_dim_padding=32)
 
-        # Fused GEMM_DQ
+        # Fused GEMM_DQ -- note we padded the input above because
+        # torch._scaled_mm is more performant for matrices with
+        # batch dimension at least 32.
```

Comment:
What is the perf effect when padding to 32 vs 16? (I ask because here it's 32 and in the PR description it's 16)

Reply:
So when I write my own wrappers for CUBLASLt, I'm getting the following error when calling the  `cublasLtMatmulAlgoGetHeuristic` with FP8:
```
[2024-05-08 19:00:00][cublasLt][1533][Info][cublasLtMatmulAlgoGetHeuristic] Unsupported M dimension for FP8 matrix multiplication. M must be divisible by 16. Got 2.
```
(this is with highest logging `CUBLASLT_LOG_LEVEL=5`) -- that's why I wrote 16 in the description.

For the setting we are using however, 32 is actually the best setting -- I tried them both and with 16 it is much closer to what it was previously. It is however possible that this will change in the future (e.g. once we use FP8 outputs I think things will change).

Reply:
I clarified this in the description now -- I wrote 16 since I didn't want to bias people for the future :)

Reply:
Hello, I have two questions:
1. I never saw the `M must be divisible by 16` error when testing cublasLt. In fact, I can perform 1 x 1024 x 16 matmul with `torch._scaled_mm`. But it seems that there are some constraints on N, cublasLt requires `N % 8 == 0`, while torch requires `N % 16 == 0`. I guess your error is also on N, because cublasLt api is col major and we pass N as M to it when using row major tensors.
2. In my experiment, matmul is slower when M is in range [1, 16], and is faster in range [17, 32], so maybe 17 is a better choice instead of 32?

Reply:
Thanks for the suggestion, let me try if `17` is better than `32` :)

Reply:
I found the performance of `17` to be exactly the same as the performance of `32`, so I'll switch to `17` since it uses less memory. Thanks for the suggestion @courage17340 :)